app/Filament/Server/Widgets/ServerConsole.php

```diff
@@ -103,6 +104,50 @@ public function enter(): void
         }
     }
 
+    public function cpuUsage(): string
+    {
+        if ($this->server->condition === 'offline') {
+            return 'Offline';
+        }
+
+        $data = collect(cache()->get("servers.{$this->server->id}.cpu_absolute"))->last() ?? 0;
+        $cpu = Number::format($data, maxPrecision: 2, locale: auth()->user()->language) . ' %';
+
+        return $cpu . ($this->server->cpu > 0 ? ' / ' . Number::format($this->server->cpu, locale: auth()->user()->language) . ' %' : ' / ∞');
+    }
+
+    public function memoryUsage(): string
+    {
+        if ($this->server->condition === 'offline') {
+            return 'Offline';
+        }
+
+        $latestMemoryUsed = collect(cache()->get("servers.{$this->server->id}.memory_bytes"))->last() ?? 0;
+        $totalMemory = collect(cache()->get("servers.{$this->server->id}.memory_limit_bytes"))->last() ?? 0;
```

Comment:
I'm not sure those actually return null, so this will work instead:

```suggestion
        $latestMemoryUsed = collect(cache()->get("servers.{$this->server->id}.memory_bytes"))->last(default: 0);
        $totalMemory = collect(cache()->get("servers.{$this->server->id}.memory_limit_bytes"))->last(default: 0);
```

Reply:
If the cache is empty it will return null, and throw a 500.

Reply:
Appears to still work with 0. Time will tell :)